packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -762,6 +762,9 @@ class FlutterPlugin implements Plugin<Project> {
     private void configurePluginProject(Map<String, Object> pluginObject) {
         assert(pluginObject.name instanceof String)
         Project pluginProject = project.rootProject.findProject(":${pluginObject.name}")
+        if (pluginProject.extensions.findByType(FlutterExtension) == null) {
```

Comment:
You are dereferencing pluginProject before checking if it is null here.

Move the check on line 768 up.

Even better, you could combine them:

if (pluginProject == null || pluginProject.extensions.findByType(FlutterExtension) != null) {
  return
}

I prefer to structure code so that it returns early rather than nesting scope. I just find that control flow easier ot reason about but it isn't strictly required.

Reply:
Whoops, thanks for catching. I moved to after the null check. 

I can't easily combine because the portions after
```
        project.dependencies {
            api(pluginProject)
        }
       ...
```
Are independent of this check

Reply:
are they though? like wouldn't the code below already have run if the extension is registered on the project?

Reply:
I guess the problem here is that I don't fully understand why some plugins already had the `FlutterExtension` created (and therefore trying again caused an error). Also, not sure this will still be a problem with the new (no new plugin) approach.

I'm going to change to remove the check to try to identify the check that fails, and fully understand.

Reply:
Ok looks like we don't need the check anymore with the new approach